plugins/catalog-backend-module-msgraph/README.md

```diff
@@ -35,6 +35,11 @@ catalog:
           # the App registration in the Microsoft Azure Portal.
           clientId: ${MICROSOFT_GRAPH_CLIENT_ID}
           clientSecret: ${MICROSOFT_GRAPH_CLIENT_SECRET_TOKEN}
+          # Optional mode for querying which defaults to "basic".
+          # By default, the Microsoft Graph API only provides the basic feature set
+          # for querying. Certain features are limited to advanced querying capabilities.
```

Comment:
So this is still useful to be able to pass in, even though we can detect the mode for `search` as advanced, right?

Reply:
yes, as certain values for "filter" will require advanced mode, too, and parsing and detecting this is way to complex and hard to maintain. Hence, we leave it to the user.

You could say we do an auto-upgrade to advanced if you use certain features like "search".

We could fail if search is used without using advanced, but that would be a breaking change.

Reply:
👌